### DISCUSSION TITLE
Why does it merge before the condition check even starts?

### DISCUSSION THREAD

**Discussion by DrAugus** [Mergifyio/mergify#5120] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/201d05b0-56b8-4eb9-87a2-9907b10ef277)
[links](https://github.com/DrAugus/draugus.github.io/pull/592)

**Comment by jd** [comment, marked as answer]
You wrote this:
![CleanShot 2024-08-05 at 11 08 01](https://github.com/user-attachments/assets/e20d2449-6f50-4c8b-b713-ad8fb121095e)

This is true when the PR is created: the PR has no checks at all, and then checks are pending, which is `!= failure`…

**Reply by DrAugus** [reply]
If this ci doesn't run, what do I write?  
like this?
```yml
and:
  - -check-success=ci
  - -check-failure=ci
  - -check-neutral=ci
  - -check-pending=ci
```

**Reply by jd** [reply]
Yes, you'll need to make sure the `ci` is not in any `check-*` status.
Though I don't think it's good idea, why not just using `check-success = deploy-life` in your case?

**Reply by DrAugus** [reply]
Because of my project, it may trigger `ci-a` to run, but not `ci-b`. I have a lot of ci's, and each time pr may be triggered by one or a few of them or all of them...

**Reply by jd** [reply]
I see. This cannot work since there's no way to know if something is wrong (Ci Is broken) or something did not run on purpose (path is ignored).

You have two choices:
1. You replicate your conditions of "run if" in Mergify: 
```
- and:
  - or:
    - files ~= ^life/
    - files ~= .github/workflows/deploy-life.yml"
    - files = package.json
    - files = pnpm-lock.yaml
  - check-success = deploy-life
```
2. Make sure each job report either skipped or success depending if it needs to run the steps. You can use https://github.com/dorny/paths-filter for example to run or not the job.

**Reply by DrAugus** [reply]
Thank you so much, it's really helpful!
